test/unit/runner.spec.js

```diff
@@ -453,7 +453,63 @@ describe('Runner', function() {
         newRunner.runTest();
       }
       expect(fail, 'to throw', 'allow unhandled errors');
-      done();
+    });
+
+    it('should not allow unhandled errors in hooks to propagate through', function() {
+      suite.beforeEach(function() {
+        throw new Error('this error will not propagate');
+      });
+      var runner = new Runner(suite);
+      expect(function() {
+        runner.hook('beforeEach', function() {});
+      }, 'not to throw');
+    });
+
+    it('should allow unhandled errors in sync hooks to propagate through', function(done) {
+      suite.beforeEach(function() {
+        throw new Error('allow unhandled errors in sync hooks');
+      });
+      var runner = new Runner(suite);
+      runner.allowUncaught = true;
+
+      var EVENT_HOOK_BEGIN = 'hook';
+      runner.on(EVENT_HOOK_BEGIN, function(hook) {
+        var _run = hook.run;
+        hook.run = function() {
+          function throwError() {
+            _run.call(hook);
+          }
+          var expected = 'allow unhandled errors in sync hooks';
+          expect(throwError, 'to throw', expected);
+          done();
+        };
+      });
+
+      runner.hook('beforeEach', function() {});
+    });
+
+    it('async - should allow unhandled errors in hooks to propagate through', function(done) {
+      // having the `done` argument triggers the async path
```

Comment:
Remove this comment.
But please add a short comment with the first test, why we need this monkey patching. In some months nobody will know otherwise.